### PR TITLE
CT-1891 ICO Appeal factories

### PR DESCRIPTION
**Pull request by aliuk2012**
- [x] adds upheld/overturned ICO appeals factories
- [x] adds upheld/overturned ICO appeals admin created cases

### CODE REVIEW

File: app/validators/closed_case_validator.rb [hunk 1]
```diff
@@ -100,14 +100,15 @@ def validate_date_ico_decision_received(rec)
...
+    elsif !rec.ico_decision.in?( Case::ICO::Base.ico_decisions.keys )
       rec.errors.add(:ico_decision, 'invalid')
     end
   end
 
   def validate_files_or_comment(rec)
     if rec.ico_decision == 'overturned' &&
-        rec.uploaded_ico_decision_files.blank? &&
+        (rec.uploaded_ico_decision_files.blank? ||
+            rec.attachments.ico_decisions.any?) &&
```
[the remark this answers is not in the document]
Comment (aliuk2012) [reply app/validators/closed_case_validator.rb]: Yes this was needed for the factories and creating valid overturned ICO Appeals

[the remark this answers is not in the document]
Comment (misaka) [reply app/validators/closed_case_validator.rb]: 👍

File: lib/cts/cases/create.rb [hunk 2]
```diff
@@ -378,6 +372,37 @@ def create_responder(responding_team)
...
+    end
+
+    def transition_to_closed_for_ico(kase)
+      kase.prepare_for_close
+      ico_decision = options.fetch(:ico_decision, Case::ICO::Base.ico_decisions.keys.sample )
+      kase.update_attributes(date_ico_decision_received: Date.today, ico_decision: ico_decision)
+      if kase.overturned?
+        uploader = S3Uploader.new(kase, CTS::dacu_manager)
+        uploader.add_file_to_case('spec/fixtures/ico_decision.png', :ico_decision)
+        # kase.attachments << FactoryBot.build(:case_ico_decision)
```
Comment (misaka) [inline lib/cts/cases/create.rb]: We should be able to drop this line now.

File: spec/factories/case/ico_fois.rb [hunk 3]
```diff
@@ -150,14 +150,25 @@
...
+      ico_decision         { "overturned" }
+      ico_decision_comment { Faker::NewGirl.quote }
+    end
+
+    after(:create) do |kase, evaluator|
+
+      if kase.ico_decision == 'overturned'
+        kase.attachments.push(*evaluator.attachments)
+        kase.save!
+      end
```
Comment (misaka) [inline spec/factories/case/ico_fois.rb]: I would definitely move this into an `after(:create)` block in the `overturned_by_ico` trait, if that works of course.

File: spec/factories/case_attachments.rb [hunk 4]
```diff
@@ -52,4 +52,12 @@
     preview_key { "#{SecureRandom.hex(16)}/request_previews/" +
                   "#{upload_group}/#{Faker::Internet.slug}.pdf" }
   end
+
+  factory :case_ico_decision, parent: :case_attachment do
+    type 'ico_decision'
```
Comment (misaka) [inline spec/factories/case_attachments.rb]: Didn't we just merge a PR to convert all such attributes from static to dynamic? (by putting them in `{...}`) (see #404)

Comment (aliuk2012) [reply spec/factories/case_attachments.rb]: Good spot!!

File: lib/cts/cases/create.rb (hunk 2 above)
Comment (aliuk2012) [reply lib/cts/cases/create.rb]: 👍

**Review by misaka — APPROVED** [review]

**Pull request merged**